app/controllers/order_item_controller.rb

```diff
@@ -0,0 +1,39 @@
+
+class OrderItemController < ApplicationController
+
+  # route for /cart
+  def index
+    @order = Order.find(session[:order_id])
+    @order_items = OrderItem.where(order_id: session[:order_id]).order("created_at asc")
+    
+  end
+
+  def create
+    # quantity 1 is default, params[:format] is where the product id comes from 
+    @order_item = OrderItem.new(order_id: session[:order_id], quantity: 1, product_id: params[:format])
+    if @order_item.save
+      redirect_to cart_path
+    else
+      redirect_to products_path
+    end
+  end
+
+  def update
+    @order_item = OrderItem.find(params[:id])
+    @order_item.update(quantity: params[:order_item][:quantity])
+
```

Comment:
For methods where you redirect instead of render, like this one, use local variables instead of instance variables. Instance variables are needed when they'll be used in the view, but there is no view here since we're responding with a redirect first.